src/common/components/elements/ListingPreview.tsx

```diff
@@ -279,7 +277,8 @@ export function ListingPreview(listing: Listing) {
   // shows up to 2 decimals, but removes pointless 0s
   const displayPrice = Number((getListingPrice(listing) * 0.000000001).toFixed(2));
 
-  if (loading) {
+  // no subdomain means it's a shell/skeleton
+  if (loading || !listing.subdomain) {
```

Comment:
hmm this a little magic logic loading logic. An argument to keep the render of the preview controlling the display of skeletons while data is being fetched.

The internal loading works because this component is responsible for fetching the information.